src/autoscaler/cf/service_plans.go

```diff
@@ -51,59 +57,48 @@ func (c *Client) getBrokerPlanGuid(ccServicePlanGuid string) (string, error) {
 	resp, err = c.httpClient.Do(req)
 
 	if err != nil {
-		logger.Error("do-request", err)
-		return "", fmt.Errorf("cf-client-get-broker-plan-guid: failed to execute request to CF API: %w", err)
+		c.logger.Error("do-request", err)
+		return nil, fmt.Errorf("cf-client-get-broker-plan-guid: failed to execute request to CF API: %w", err)
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
 		err = fmt.Errorf("cf-client-get-broker-plan-guid: failed to get service plan: %s [%d] %s", servicePlansUrl.String(), resp.StatusCode, resp.Status)
-		logger.Error("get-response", err)
-		return "", err
+		c.logger.Error("get-response", err)
+		return nil, err
 	}
 
 	result := &ServicePlanResource{}
 
 	err = json.NewDecoder(resp.Body).Decode(result)
 	if err != nil {
-		logger.Error("decode", err)
-		return "", fmt.Errorf("cf-client-get-broker-plan-guid: failed to decode response from CF API: %w", err)
+		c.logger.Error("decode", err)
+		return nil, fmt.Errorf("cf-client-get-broker-plan-guid: failed to decode response from CF API: %w", err)
 	}
-
-	brokerPlanGuid := result.Entity.UniqueId
-	logger.Info("found-guid", lager.Data{"brokerPlanGuid": brokerPlanGuid})
-	c.ccServicePlanToBrokerPlanGuid[ccServicePlanGuid] = brokerPlanGuid
-	c.brokerPlanGuidToCCServicePlanGuid[brokerPlanGuid] = ccServicePlanGuid
-
-	return brokerPlanGuid, nil
+	return result, nil
 }
 
+/*GetServicePlan
+ * v3: GetServicePlan(serviceInstanceGuid) -> (GET /v3/service_instance/{serviceInstanceGuid}) -> (GET /v2/service_plan/{.Relationships.ServicePlan.Data.Guid}).Entity.UniqueId
+ * v2: (get /v2/service_instances/serviceInstanceGuid) -> (GET /v2/service_plan/{.Entity.ServicePlanGuid}).Entity.UniqueId == "name-id"
+ */
 func (c *Client) GetServicePlan(serviceInstanceGuid string) (string, error) {
-	logger := c.logger.Session("cf-client-get-service-plan", lager.Data{"serviceInstanceGuid": serviceInstanceGuid})
-	logger.Debug("start")
-	defer logger.Debug("end")
-
-	c.instanceMapLock.Lock()
```

Comment:
BTW: 
This caching was very inefficient in that it locks it for every read or write for upto the lenght of the https call (2-3 seconds). 
The memoiser only locks it for writes if its not already found otherwise it has many reads concurrently.